SingularityService/src/main/java/com/hubspot/singularity/hooks/SingularityWebhookSender.java

```diff
@@ -60,36 +61,33 @@ public SingularityWebhookSender(SingularityConfiguration configuration,
     this.taskHistoryHelper = taskHistoryHelper;
     this.objectMapper = objectMapper;
 
-    this.webhookSemaphore = AsyncSemaphore.newBuilder(configuration::getMaxConcurrentWebhooks, executorServiceFactory.get("webhook-semaphore", 5)).build();
+    this.webhookSemaphore = AsyncSemaphore.newBuilder(configuration::getMaxConcurrentWebhooks, executorServiceFactory.get("webhook-semaphore", 1)).build();
   }
 
   public void checkWebhooks() {
     final long start = System.currentTimeMillis();
 
-    final List<SingularityWebhook> webhooks = webhookManager.getActiveWebhooks();
-    if (webhooks.isEmpty()) {
-      return;
-    }
-
     int taskUpdates = 0;
     int requestUpdates = 0;
     int deployUpdates = 0;
 
     List<CompletableFuture<Response>> webhookFutures = new ArrayList<>();
 
-    for (SingularityWebhook webhook : webhooks) {
-      switch (webhook.getType()) {
-        case TASK:
-          taskUpdates += checkTaskUpdates(webhook, webhookFutures);
-          break;
-        case REQUEST:
-          requestUpdates += checkRequestUpdates(webhook, webhookFutures);
-          break;
-        case DEPLOY:
-          deployUpdates += checkDeployUpdates(webhook, webhookFutures);
-          break;
-        default:
-          break;
+    for (WebhookType type : WebhookType.values()) {
+      for (SingularityWebhook webhook : webhookManager.getActiveWebhooksByType(type)) {
```

Comment:
this is a switch to use the cached version of this data since it can be queried frequently